cog/tasks/fixme.py

```diff
@@ -7,7 +7,8 @@
 class FIXMECheck(cog.task.Task):
     '''Find the word "fixme" in the code.'''
     def __init__(self, *args):
-        cog.task.Task.__init__(self, *args)
+        if args:
```

Comment:
What is this (and related) change for? `args` should be an argument list, so is this the same as `if len(args) > 0`? Are you sure no tasks have zero length argument lists and should still call the super init?

Reply:
Yes, it is the same. In fact, `if args` is the more "Pythonic" way to check whether or not a list (or other sequences) is empty. Using checkers like Pylint will highlight this. See [here](https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty) for a discussion I found (although as someone there pointed out, it is also less explicit).

As for what the change is for, it is so that one can run the task on their own computer before submitting a PR. This change simplifies the process a bit so that arguments to the constructor don't need to be passed in. Then, one can check a task like:

```python
#!/usr/bin/python2

from cog.tasks.chartest import CharCheck

p = CharCheck()

# First argument: your repository.
# Second argument: commit hash.
# The other two arguments are the repository and branch being merged in to.
doc = {'kwargs': {'git_url': 'git@github.com:mark-r-anderson/rat.git',
                  'sha': 'ede2949553da7b86527b5a77341ed1fa0d6f490e',
                  'base_repo_url': 'git@github.com:snoplus/rat.git',
                  'base_repo_ref': 'master'}}

# Second argument here is where you want to clone the repository to.
p.run(doc, '~/builds')
```

And yes, before if you passed no arguments to the constructor of the task, it would fail. The \_\_init\_\_ function of Task requires a certain number of arguments and has no defaults:

https://github.com/snoplus/cog/blob/49f17fc83bd3ad7fe73afb080ca5dccbf38e95e7/cog/task.py#L23-L27

So this should not affect any existing tasks. They would've failed if no arguments were supplied.

Reply:
I also realized I didn't quite answer your last question here -- as long as one is only calling the run method of a given task, then the super init does not need to be called. This is only for user testing, not for the on demand builds.

Reply:
My main gripe with this change was that it isn't obvious what's being accomplished. It would be better to modify the `Task.__init__` function to accept `*args` and only expand `args` to `host,dbname,username,password,doc_id`  and connect to couch if it's nonzero length. That way you don't have to change every subclass to gain this functionality, and you're not optionally calling the super init based on arguments passed.

Reply:
OK, that's a better idea. I'll do that instead